docs/spec.md

```diff
@@ -1048,23 +1048,25 @@ def batch_norm_grad(operand, scale, mean, variance, grad_output, epsilon, featur
   # Use the implementation from batchnorm_expander.cc in XLA
   # Temporary variables have exactly the same names as in the C++ code
   elements_per_feature = constant(
-    divide(size(operand), dim(operand, feature_index)))
+    divide(num_elements(operand), dim(operand, feature_index)))
   i1 = multiply(
     grad_output,
     broadcast_in_dim(elements_per_feature, [], shape(operand)))
   i2 = broadcast_in_dim(
     compute_sum(grad_output, feature_index),
     [feature_index], shape(operand))
   i3 = broadcast_in_dim(
-    compute_sum(multiply(grad_output, centered_operand)),
+    compute_sum(multiply(grad_output, centered_operand), feature_index),
     [feature_index], shape(operand))
   i4 = multiply(i3, centered_operand)
   i5 = divide(i4, add(variance_bcast, epsilon_bcast))
+  i6 = subtract(subtract(i1, i2), i5)
   grad_operand = multiply(
     divide(divide(scale_bcast, stddev), elements_per_feature),
```

Comment:
Looks like elements_per_feature needs to be broadcast. Let's do that at the definition site?